geomstats/_backend/pytorch/__init__.py

```diff
@@ -780,6 +777,18 @@ def vectorize(x, pyfunc, multiple_args=False, **kwargs):
     return stack(list(map(pyfunc, x)))
 
 
+def vec_to_diag(vec):
```

Comment:
Why not: https://pytorch.org/docs/1.8.1/generated/torch.diag_embed.html?highlight=diag#torch.diag_embed ?

In this case, this would go on Line 44 as:
vec_to_diag = torch.diag_embed